velox/exec/Window.cpp

```diff
@@ -284,8 +284,13 @@ void updateKRowsOffsetsColumn(
   // moves ahead.
   int precedingFactor = isKPreceding ? -1 : 1;
   for (auto i = 0; i < numRows; i++) {
-    rawFrameBounds[i] =
-        (startRow + i) + vector_size_t(precedingFactor * offsets[i]);
+    auto startValue = (int64_t)(startRow + i) + precedingFactor * offsets[i];
+    // Considers integer overflow case.
+    if (startValue != (int32_t)startValue) {
```

Comment:
Encapsulate this in an inline function like isIntegerOverflow or fixBoundForOverflow and re-use the function in both the constant and column overflow cases.

Reply:
The handling for constant preceding/following is different from column now. Maybe, we can just keep this part unchanged. Thanks!